example/v1/file-upload.ts

```diff
@@ -0,0 +1,24 @@
+import {z, defaultEndpointsFactory} from '../../src';
+import crypto from 'crypto';
+
+export const fileUploadEndpoint = defaultEndpointsFactory.build({
+  method: 'post',
+  type: 'upload',
```

Comment:
can we restrict `type=upload` to `post` method only?

Reply:
no, but perhaps we can detect the type by usage of `z.upload()` within `input` in the future